src/Hosting/Hosting/src/Internal/HostingApplication.cs

```diff
@@ -53,16 +65,87 @@ public void DisposeContext(Context context, Exception exception)
             _diagnostics.RequestEnd(httpContext, exception, context);
             _httpContextFactory.Dispose(httpContext);
             _diagnostics.ContextDisposed(context);
+
+            context.Reset();
         }
 
+        // Struct to turn {TContext} uses into faster concrete generic implementations rather than shared generics.
         public struct Context
         {
-            public HttpContext HttpContext { get; set; }
-            public IDisposable Scope { get; set; }
-            public long StartTimestamp { get; set; }
-            public bool EventLogEnabled { get; set; }
-            public Activity Activity { get; set; }
-            internal bool HasDiagnosticListener { get; set; }
+            // Single field wrapper over an object to keep pass-by-value sematics in a single pointer register
+            // rather than involving stack copies.
+            private InnerContext _context;
+
+            internal static Context Create()
+            {
+                return new Context() { _context = new InnerContext() };
```

Comment:
Is this only only place a new Context is created? If so, why have the Initialize method?

Reply:
If `IHostContextContainer<Context>` is implemented it never calls `Create()` and only calls `Initialize()`

Reply:
Mainly I thought having `Context.Create()` was less strange then `deafult(Context).Initialize()`

Suppose could do `(new Context()).Initialize()` but again seems messy?

Reply:
I see. I missed that Http1Connection/Http2Stream aren't calling Create when implementing `IHostContextContainer<Context>`. I think it's fine as is then.